engine/env/spark/src/test/java/org/datacleaner/spark/JobResultPathTest.java

```diff
@@ -0,0 +1,152 @@
+ * or otherwise. The default setting for 'fs.defaultFS' is 'file:///'
+ */
+public class JobResultPathTest extends TestCase {
+
+    @Test
+    public void testResultJobNameCorrectSpecified() throws Exception {
+        final SparkConf sparkConf = new SparkConf().setMaster("local").setAppName("DCTest - " + getName());
+        final JavaSparkContext sparkContext = new JavaSparkContext(sparkConf);
+        try {
+
+            final SparkJobContext sparkJobContext = new SparkJobContext(sparkContext,
+                    "src/test/resources/conf_local.xml", "src/test/resources/vanilla-job.analysis.xml",
+                    "src/test/resources/jobProperties/job.properties");
+            final AnalysisJob job = sparkJobContext.getAnalysisJob();
+            assertNotNull(job);
+            assertEquals("hdfs://bigdatavm:9000/target/results/myresult.analysis.result.dat",
+                    sparkJobContext.getResultPathUsercustomized());
+            assertEquals("vanilla-job", sparkJobContext.getAnalysisJobName());
+            assertEquals("hdfs://bigdatavm:9000/target/results/myresult.analysis.result.dat",
+                    sparkJobContext.getResultJobFilePath());
+        } finally {
+            sparkContext.close();
+        }
+    }
+
+    @Test
+    public void testResultJobNameSpecifiedEmpty() throws Exception {
+        final SparkConf sparkConf = new SparkConf().setMaster("local").setAppName("DCTest - " + getName());
+        final JavaSparkContext sparkContext = new JavaSparkContext(sparkConf);
+        try {
+
+            final SparkJobContext sparkJobContext = new SparkJobContext(sparkContext,
+                    "src/test/resources/conf_local.xml", "src/test/resources/vanilla-job.analysis.xml",
+                    "src/test/resources/jobProperties/job1.properties");
+            final AnalysisJob job = sparkJobContext.getAnalysisJob();
+            assertNotNull(job);
+            assertEquals("", sparkJobContext.getResultPathUsercustomized());
+            final String analysisJobName = sparkJobContext.getAnalysisJobName();
+            assertEquals("vanilla-job", analysisJobName);
+
+            final String resultJobFilePath = sparkJobContext.getResultJobFilePath();
+            assertTrue(resultJobFilePath.contains(analysisJobName));
+
+        } finally {
+            sparkContext.close();
+        }
+    }
+
+    @Test
+    public void testResultJobNameJustFolderSpecified() throws Exception {
+        final SparkConf sparkConf = new SparkConf().setMaster("local").setAppName("DCTest - " + getName());
+        final JavaSparkContext sparkContext = new JavaSparkContext(sparkConf);
+        try {
+
+            final SparkJobContext sparkJobContext = new SparkJobContext(sparkContext,
+                    "src/test/resources/conf_local.xml", "src/test/resources/vanilla-job.analysis.xml",
+                    "src/test/resources/jobProperties/job2.properties");
```

Comment:
could we name all these properties file with some semantics instead of numbers? Right now it is pretty tough to pair a properties file with a test case in the reader's mind.